data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurer.java

```diff
@@ -108,18 +112,38 @@ public class KafkaSecurityConfigurer {
         properties.put(SASL_JAAS_CONFIG, String.format(PLAINTEXT_JAASCONFIG, username, password));
     }*/
 
-    private static void setPlainTextAuthProperties(Properties properties, final PlainTextAuthConfig plainTextAuthConfig, EncryptionType encryptionType) {
-        String username = plainTextAuthConfig.getUsername();
-        String password = plainTextAuthConfig.getPassword();
+    private static void setPlainTextAuthProperties(final Properties properties, final PlainTextAuthConfig plainTextAuthConfig,
+                                                   final EncryptionConfig encryptionConfig) {
+        final String username = plainTextAuthConfig.getUsername();
+        final String password = plainTextAuthConfig.getPassword();
         properties.put(SASL_MECHANISM, "PLAIN");
         properties.put(SASL_JAAS_CONFIG, "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"" + username + "\" password=\"" + password + "\";");
-        if (encryptionType == EncryptionType.NONE) {
-            properties.put(SECURITY_PROTOCOL, "SASL_PLAINTEXT");
-        } else { // EncryptionType.SSL
+        if (checkEncryptionType(encryptionConfig, EncryptionType.SSL)) {
             properties.put(SECURITY_PROTOCOL, "SASL_SSL");
+            setSecurityProtocolSSLProperties(properties, encryptionConfig);
+        } else { // EncryptionType.NONE
+            properties.put(SECURITY_PROTOCOL, "SASL_PLAINTEXT");
         }
```

Comment:
Why switch the statements here?

Reply:
`encryptionType` can be null, so had added `checkEncryptionType` method that does null check. 

I could have kept the same order with if condition `if (checkEncryptionType(encryptionConfig, EncryptionType.NONE)) {`, but since we are handling addition things for SSL, I reversed the order.